build.gradle

```diff
@@ -51,17 +51,31 @@ tasks.withType(JavaCompile) {
   options.compilerArgs << "-Xlint:unchecked" << "-Werror"
 }
 
-task functional(type: Test, description: 'Runs the functional tests.', group: 'Verification') {
+tasks.withType(Test) {
+  useJUnitPlatform()
+
+  testLogging {
+    exceptionFormat = 'full'
+  }
+}
+
+test {
+  failFast = true
```

Comment:
is failFast what we want? i.e. do you want to know a test is broken or all of the tests that are broken?

Reply:
unit tests should be run by developer locally. they are not time consuming or smth. if any failures occur - should be addressed as fast as possible. not wait for full suite finish. plus, if one is developing a feature he/she must be aware of the changes and perhaps track those errors faster

another pro: pipeline will fail faster if we tend to be lazy and not run those commands locally before pushing code for jenkins to do the job

this goes for integration tests mostly as those are time and resource consuming. if one fails - why bother run them all. what if you just forgot to spin up docker and have 50-100 testcontainers test..